addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/AbstractNetatmoThingHandler.java

```diff
@@ -156,4 +157,18 @@ protected String getId() {
         }
     }
 
+    protected void updateProperties(Integer firmware, String modelId) {
+        Map<String, String> properties = editProperties();
+        if ((firmware != null || (modelId != null)) && (properties.get(Thing.PROPERTY_VENDOR) == null)) {
+            updateProperty(Thing.PROPERTY_VENDOR, VENDOR);
+        }
+        if ((firmware != null) && ((properties.get(Thing.PROPERTY_FIRMWARE_VERSION) == null)
+                || !properties.get(Thing.PROPERTY_FIRMWARE_VERSION).equals(firmware.toString()))) {
```

Comment:
https://github.com/eclipse/smarthome/blob/5b746e3084ca2803400549bd5467a11c7116d92b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java#L598

The `updateProperty` method checks the last two conditions on its own. They are redundant in this case.

Reply:
Good to know. I will suppress the checks.